packages/autocomplete-js/src/autocomplete.ts

```diff
@@ -341,10 +341,21 @@ export function autocomplete<TItem extends BaseItem>(
   function update(updatedOptions: Partial<AutocompleteOptions<TItem>> = {}) {
     cleanupEffects();
 
+    const { components, ...rendererProps } = props.value.renderer;
+
     optionsRef.current = mergeDeep(
-      props.value.renderer,
+      rendererProps,
       props.value.core,
-      { initialState: lastStateRef.current },
+      {
+        // We need to filter out default components so they can be replaced with
+        // a new `renderer`, without getting rid of user components.
+        // @MAJOR Pass all `components` and let default ones be overriden.
```

Comment:
I'm really not in favor of adding arbitrary "major" instructions in the code that weren't discussed elsewhere beforehand. It won't empower us to take another API decision when it's time to take it. Adding a "maybe" leaves us more room for the future.

Reply:
I added them so we could discuss it in the PR, and because this is typically what we would look for when making a new major. I'm fine with another convention as long as we don't forget them.

Reply:
Reworded here, wdyt? 5a5de5f3

Reply:
I'm not against the `@MAJOR`. I'm against proposing a fixed solution in a comment.

The rewording is good.